scanpy/utils.py

```diff
@@ -667,8 +667,22 @@ def select_groups(adata, groups_order_subset='all', key='groups'):
     if groups_order_subset != 'all':
         groups_ids = []
         for name in groups_order_subset:
-            groups_ids.append(
-                np.where(adata.obs[key].cat.categories.values == name)[0][0])
+            # deal with case that all group names were originally integers,
```

Comment:
All of this is a lot of code that doesn't handle Scanpy's deprecation of non-string categories in a good way. So, I don't like to merge this.

Instead, we should print a warning if someone operates with non-string categories. If you just make it a warning, here, `logg.warn`, that's fine.